test/e2e/providers/machineset/machineset.go

```diff
@@ -10,8 +10,12 @@ import (
 )
 
 // New returns a new MachineSet for use with the e2e test suite
-func New(rawProvider []byte, infrastructureName string, replicas int32, withIgnoreLabel bool) *mapi.MachineSet {
-	machineSetName := machineSetName(withIgnoreLabel)
+func New(rawProvider []byte, infrastructureName string, replicas int32, withIgnoreLabel bool, useInfraName bool) *mapi.MachineSet {
```

Comment:
The proposed param `useInfraName` is redundant with `infrastructureName`.  I suggest using a string param to capture the desired prefix for the machineSet, e.g. `namePrefix` and you pass the infrastructure name when invoking the function. For example:

```go
    machineset.New(rawSpec, p.InfrastructureName, replicas, withIgnoreLabel, p.InfrastructureName)
```

Reply:
Done.